### DISCUSSION TITLE
How and where should I update content in CV?

### DISCUSSION THREAD

**Discussion by Yuanhao-JIANG** [alshedivat/al-folio#1611] [Q&A] [answered]
I want to change content in CV page, also delete the table of content.

But I notice there are
_layout/cv.html
_data/cv.yml
assets/json/resume.json

are these the right places to change? if so, what is the workflow for updating cv? (seems like change _data/cv.yml have no effect and changing assets/json/resume.json has some effects)

**Comment by george-gca** [comment, marked as answer]
`_layouts/cv.html` is responsible for rendering the contents of the cv. If you check [here](https://github.com/alshedivat/al-folio/blob/c824d128673c000ef9f0e0e77a0ae30d2c4b4568/_layouts/cv.html#L5) there is an `unless` condition with `site.data.resume`. Everything that is accessed via `site.` is defined in the `_config.yaml`. In this case, this information is defined [here](https://github.com/alshedivat/al-folio/blob/c824d128673c000ef9f0e0e77a0ae30d2c4b4568/_config.yml#L413).

What this means is, if there is no `resume` data defined in `_config.yml` and loaded via a json file, it will load the contents of `_data/cv.yml` as fallback.

To remove the table of contents, simply remove the referred part from `_layouts/cv.html`.

**Reply by Weixiang-Sun** [reply]
U mean remove the whole sentence or just remove "unless", I try both but it failed to deploy.

**Reply by Yuanhao-JIANG** [reply]
remove the 'Table of contents' heading and the following unordered list in _layouts/cv.html

**Reply by xiong-zx** [reply]
I have noticed that there are many files and directories related to "CV", including:
- _data/cv_.yml
- _includes/cv/
- _includes/resume/
- pages/cv.md

What are the priorities of these files and directories? I suppose that there should be only one place to modify the cv.

**Reply by george-gca** [reply]
What do you mean by modifying the cv?

The only files responsible for the content of the cv are:

- data/cv.yml
- assets/json/resume.json

All the others are responsible for generating the needed html for visualizing the information.

The main one is `assets/json/resume.json` and it was introduced in #1547. It is a known standard for creating cv content in json format, so you could consume this data from various parsers and generate your cv programmatically. 

As for `data/cv.yml` it was the original way of creating the content for the cv page, and it is currently used as a fallback if the file `assets/json/resume.json` is not found. Since it was already implemented and is more human readable than a json file we decided to keep it as an option.

**Reply by chriskcovert** [reply]
are you saying that if you comment out (#) the sections like such it won't show that section in the cv?

jsonresume:
  - basics
  - work
  - education
  #- publications
  - projects
 # - volunteer
  - awards
  - certificates
  - skills
  - languages
  - interests
  #- references
I tried removing the section completely from the resume.json file but it would fail the validation(I think) and revert to the cv.yml.
requesting clear instructions on how to hide/remove sections of the CV using the JSON file and _config.yml file.
Can you please clarify?
